x-pack/plugins/kubernetes_security/public/components/percent_widget/hooks.ts

```diff
@@ -20,27 +20,19 @@ export const useFetchPercentWidgetData = (
 ) => {
   const { http } = useKibana<CoreStart>().services;
   const cachingKeys = [QUERY_KEY_PERCENT_WIDGET, widgetKey, filterQuery, groupBy, countBy, index];
-  const query = useQuery(
-    cachingKeys,
-    async (): Promise<Record<string, number>> => {
-      const res = await http.get<AggregateResult[]>(AGGREGATE_ROUTE, {
-        query: {
-          query: filterQuery,
-          groupBy,
-          countBy,
-          page: 0,
-          index,
-        },
-      });
+  const query = useQuery(cachingKeys, async (): Promise<Record<string, number>> => {
+    const res = await http.get<AggregateResult>(AGGREGATE_ROUTE, {
+      query: {
+        query: filterQuery,
+        groupBy,
+        countBy,
+        page: 0,
+        index,
+      },
+    });
 
-      return onReduce(res);
-    },
-    {
-      refetchOnWindowFocus: false,
```

Comment:
are we sure we want to refetch on window focus? Could result in many unnecessary queries?

Reply:
not at all, that's why I removed it from here and added it as a default option at the top `queryClient` level, so this will be the default behaviour for the Kubernetes plugin

```js
// Initializing react-query
const queryClient = new QueryClient();
const queryClient = new QueryClient({
  defaultOptions: {
    queries: {
      refetchOnWindowFocus: false,
      refetchOnMount: false,
      refetchOnReconnect: false,
    },
  },
});
```

https://github.com/opauloh/kibana/blob/k8s_tree_view/x-pack/plugins/kubernetes_security/public/methods/index.tsx